features/netsocket/emac-drivers/TARGET_STM/stm32xx_emac.cpp

```diff
@@ -546,7 +546,7 @@ int STM32_EMAC::low_level_input(emac_mem_buf_t **buf)
 
     dmarxdesc = EthHandle.RxFrameInfos.FSRxDesc;
 
-    if (len > 0) {
+    if (len > 0 && EthHandle.RxFrameInfos.length <= 1500) {
```

Comment:
As @jeromecoutant  also noted, the data-types do not match.

the len probably could be changed to uint32_t (I presume the compiler is already warning about this?).

I wonder why are we even getting that large packets as that would be much larger than any MTU allows?

Still, this at least fixes the original issue. The BR no longer hangs with this change.

Reply:
This just does not look like the root-cause, more like symptom.

Reply:
Shouldn't be hard-coding 1500 here - should be using whatever you know the size of your actual buffer is, which could be smaller or larger. It's a validity check that the descriptor is consistent with the buffer.

Reply:
@teetak01 , I completely agree, this is not fixing the root cause and the PR was made to test if ignoring the extremely long packet will help to the Nanostack jam problem. It did and now the driver folks should find the root cause and make a proper fix.

@kjbracey-arm , when the error occurred, the buffer content was valid.  Just the provided `len` was incorrect and it was causing problem in Nanostack buffer handling that will be fixed in the next Mbed OS version.